File: BYTES.NET/WPF/Converters/NothingToVisibilityConverter.vb
'import .net namespace(s) needed
Imports System.Globalization

Namespace WPF.Converters

    ''' <summary>
    ''' converter returning the visibility state based on an object being nothing or not nothing
    ''' </summary>
    ''' <remarks>'Visible' will be returned by default, if value is 'Nothing'; if parameter is set to 'false', 'Visible' will be returned if value is not 'Nothing'; if para´meter is set to 'true', strings of length 0 will be handled like 'Nothing'</remarks>
    Public Class NothingToVisibilityConverter

        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert

            If IsNothing(parameter) OrElse String.IsNullOrEmpty(parameter) Then 'no parameter was given

                If IsNothing(value) Then

                    Return Visibility.Visible

                Else

                    Return Visibility.Collapsed

                End If

            Else 'a parameter was given

                Try

                    Dim comparing As Boolean = CBool(parameter)

                    If comparing Then

                        If IsNothing(value) = comparing OrElse String.IsNullOrEmpty(value) Then

                            Return Visibility.Visible

                        Else

                            Return Visibility.Collapsed

                        End If

                    Else

                        If Not IsNothing(value) Then

                            Return Visibility.Visible

                        Else

                            Return Visibility.Collapsed

                        End If

                    End If

                Catch ex As Exception

                    Throw New ArgumentException("Unable to convert value")

                End Try

            End If

        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack

            Return Binding.DoNothing

        End Function

    End Class

End Namespace

File: src-2023/BYTES.NET.WPF/Converters/NothingToVisibilityConverter.cs
//import .net namespace(s) required
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Data;

namespace BYTES.NET.WPF.Converters
{
    public class NothingToVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if(parameter == null)
            {
                parameter = true;
            }

            if ((bool)parameter)
            {
                if(value == null)
                {
                    return Visibility.Visible;
                } else
                {
                    if(value.GetType() == typeof(string))
                    {
                        if (string.IsNullOrEmpty((string)value) || string.IsNullOrWhiteSpace((string)value))
                        {
                            return Visibility.Visible;
                        }
                    }

                    return Visibility.Collapsed;
                }

            } else
            {
               
                if (value == null)
                {
                    return Visibility.Collapsed;
                }
                else
                {
                    if (value.GetType() == typeof(string))
                    {
                        if (string.IsNullOrEmpty((string)value) || string.IsNullOrWhiteSpace((string)value))
                        {
                            return Visibility.Collapsed;
                        }
                    }

                    return Visibility.Visible;
                }
            }

        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return Binding.DoNothing;
        }
    }
}
